File: cp35/cpp/anagram.cpp
    #include<bits/stdc++.h>
    using namespace std;  
    
    bool isAnagram(string a, string b){
        vector<char> chars(26);
        int str1 = a.length();
        int str2 = b.length();
        if(str1 != str2)
            return false;
        for(int i = 0; i < str1; i++){
            chars[a[i]-'a']++;
            chars[b[i]-'a']--;
        }
        
        for(int i = 0; i < chars.size(); i++){
            if(chars[i] != 0)
                return false;
        }
        
        return true;
        
    };

    int main () {
        string a = "abcd";
        string b = "dcba";
        cout << isAnagram(a, b) << endl;
        return 0;
    }

File: cp35/python/anagram.py
def isAnagram(a, b):
    chars = [0] * 26
    str1 = len(a)
    str2 = len(b)
    if str1 != str2:
        return False
    for i in range(str1):
        chars[ord(a[i]) - ord('a')] += 1
        chars[ord(b[i]) - ord('a')] -= 1
    for i in range(len(chars)):
        if chars[i] != 0:
            return False
    return True


def main():
    a = "abcd"
    b = "dcba"
    print(isAnagram(a, b))


if __name__ == "__main__":
    main()